static/js/AdminEditor.jsx

```diff
@@ -177,6 +180,15 @@ const AdminEditor = ({title, data, close, catMenu, pictureUploader, updateData,
                 obj = <textarea className="default" id={field} onChange={setInputValue} defaultValue={data[field]}
                          placeholder={Sefaria._(placeholder)}/>;
                 break;
+            case 'textarea readonly':
+                obj = <textarea readOnly  className="default" id={field} onChange={setInputValue} defaultValue={data[field]}
```

Comment:
the following two cases duplicate a lot of code. Instead create a component that takes `readOnly` as a parameter and reuse this component for both textarea cases. Do the same for the readonly input.